### PR TITLE
Feature/#6 Create charlist skeleton page

### PR DESCRIPTION
**Pull request by FFreestyler**

### CODE REVIEW

File: src/components/MainPageComponent.tsx [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
   const theme = useTheme();
+  const boxSx = {
+    border: '4px solid',
+    borderColor: theme.palette.primary.main,
+    backgroundColor: theme.palette.secondary.main,
+    borderRadius: '4px',
+    padding: '1rem',
+    mb: '1rem',
+  };
+
```
Comment (MangriMen) [inline src/components/MainPageComponent.tsx]: I think we need styled box with this sx as separate component, except margin bottom of course, and padding maybe.

File: src/components/MainPageComponent.tsx [hunk 2]
```diff
@@ -11,10 +27,105 @@ function MainPageComponent() {
         border: '4px solid',
         borderColor: theme.palette.primary.main,
         borderRadius: '4px',
```
Comment (MangriMen) [inline src/components/MainPageComponent.tsx:29]: The same as with box, i think need styled

Comment (MangriMen) [inline src/components/MainPageComponent.tsx]: The same as with box, i think need styled

**Review by MangriMen — CHANGES_REQUESTED** [review]

**Comment by MangriMen** [comment]
[Create charlist page skeleton](https://trello.com/c/2wac8trU/6-create-charlist-page-skeleton)

File: src/components/MainBoxStyled.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Box, BoxProps, styled } from '@mui/material';
+
+const BoxStyled = styled(Box)(({ theme }) => ({
+  border: '4px solid',
+  borderColor: theme.palette.primary.main,
+  backgroundColor: theme.palette.secondary.main,
+  borderRadius: '4px',
+  padding: '1rem',
```
Comment (MangriMen) [inline src/components/MainBoxStyled.tsx]: Padding is not needed here

File: src/components/MainBoxStyled.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+}));
+
+function MainBoxStyled({ children, ...props }: BoxProps) {
+  return <BoxStyled {...props}>{children}</BoxStyled>;
+}
+
+export default MainBoxStyled;
```
Comment (MangriMen) [inline src/components/MainBoxStyled.tsx]: Naming for common styled is ParComponentName

**Review by MangriMen — CHANGES_REQUESTED** [review]
Check comments

**Review by MangriMen — APPROVED** [review]
Good

File: src/components/MainPageComponent.tsx [hunk 5]
```diff
@@ -11,10 +27,105 @@ function MainPageComponent() {
...
+            <Grid container height="100%" flexDirection="column">
+              {Array.from(Array(17)).map((_, index) => (
+                <Grid item key={index}>
+                  <Grid
+                    container
+                    flexDirection="row"
+                    justifyContent="space-between"
+                  >
+                    {Array.from(Array(3)).map((_, index) => (
+                      <Box sx={{ ...boxSx }} key={index}></Box>
```
Comment (IHiFiveI) [inline src/components/MainPageComponent.tsx]: Remove useless copying of styles object, as already done on line 56 (same goes for every other `sx={{...boxSx}}` on page)

Comment (IHiFiveI) [inline src/components/MainPageComponent.tsx]: `key={index}` is bad. I see, you can't do anything about it here, but maybe consider using `uuidv4` lib?

File: src/components/MainPageComponent.tsx (hunk 1 above)
Comment (IHiFiveI) [reply src/components/MainPageComponent.tsx]: > I think we need styled box with this sx as separate component, except margin bottom of course, and padding maybe.

Yes, we do

File: src/components/MainPageComponent.tsx (hunk 5 above)
Comment (IHiFiveI) [reply src/components/MainPageComponent.tsx]: My bad, was covered on [this comment](https://github.com/MangriMen/Paramoios/pull/35/files/00f024d12990bca29519566ea42781b00f7f4109#r874176648)

Comment (FFreestyler) [reply src/components/MainPageComponent.tsx]: Ok, but I think in future we dispose of components in array.

**Review by IHiFiveI — APPROVED** [review]

**Pull request merged**